### ISSUE TITLE
[minor cleanup]: Use different variable for flux_msg_handler_t pointers

### ISSUE THREAD

**Issue by chu11** [flux-framework/flux-core#1219]
I got confused looking at some code in ```msg_handler.c``` and ```reactor.c``` until I realized ```w``` is used for ```flux_msg_handler_t``` pointers and also for ```flux_watcher_t``` pointers.

Looking back, ```flux_msg_handler_t``` was once ```flux_msg_watcher_t```, thus the legacy naming.  We could update it.

**Comment by garlick** [comment]
Works for me.  Sorry for my laziness.

**Comment by chu11** [comment]
It's worth mentioning if ```struct flux_msg_handler_spec``` is updated to change ```w``` to something else, that'll incur changes in flux sched as well.

**Comment by garlick** [comment]
The common use case is to statically initialize
```c
static struct flux_msg_handler_spec sim_htab[] = {
    {FLUX_MSGTYPE_EVENT, "sim.start", start_cb, 0, NULL},
    {FLUX_MSGTYPE_REQUEST, "sched.trigger", trigger_cb, 0, NULL},
    {FLUX_MSGTYPE_EVENT, "sched.res.*", sim_res_event_cb, 0, NULL},
    FLUX_MSGHANDLER_TABLE_END,
};
```
and then pass to addvec/delvec
```c
   if (flux_msg_handler_addvec (ctx->h, sim_htab, (void *)h) < 0) {
        flux_log (ctx->h, LOG_ERR, "flux_msg_handler_addvec: %s", strerror (errno));
        goto done;
    }
```
so possibly the name of that structure member won't be a problem in sched.

**Comment by garlick** [comment]
This should be closed when #1277 is merged, since it makes the `w` member go away.

**Comment by garlick** [comment]
This should be closed when #1277 is merged, since it makes the `w` member go away.

**Comment by chu11** [comment]
it makes the ```w``` member go away in the handler spec, but I think it's still used as ```w``` in a lot of places (function parameters, etc.).
